br/pkg/storage/s3.go

```diff
@@ -666,6 +677,18 @@ func (r *s3ObjectReader) Seek(offset int64, whence int) (int64, error) {
 
 	if realOffset == r.pos {
 		return realOffset, nil
+	} else if realOffset >= r.rangeInfo.Size {
+		// See: https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.35
+		// because s3's GetObject interface doesn't allow get a range that matches zero length data,
+		// so if the position is out of range, we need to always return io.EOF after the seek operation.
+
+		// close current read and open a new one which target offset
+		if err := r.reader.Close(); err != nil {
+			log.L().Warn("close s3 reader failed, will ignore this error", logutil.ShortError(err))
+		}
+
+		r.reader = eofReader{}
+		return r.rangeInfo.Size, nil
```

Comment:
`r.pos` need to be updated in case subsequent  seeks. Also, should we return `realOffset` rather than `r.rangeInfo.Size`? I didn't find any specifications for this behavior, but returning `realOffset` is more consistent with linux filesystem.

Reply:
Fixed. I think we should return real position which is `r.rangeInfo.Size`, since the behavior isn't the same as linux fs. fs implement `ReaderWriter` and allow to write to a position that is larger that the current end and leave the file with a hole. But here we don't. Though there is no much difference.